File: app/src/main/java/at/shockbytes/dante/network/BookDownloader.java
package at.shockbytes.dante.network;

import at.shockbytes.dante.util.books.BookSuggestion;
import io.reactivex.Observable;

/**
 * @author Martin Macheiner
 *         Date: 13.02.2017.
 */

public interface BookDownloader {

    Observable<BookSuggestion> downloadBookSuggestion(String isbn);

}

File: app/src/main/java/at/shockbytes/dante/network/BookDownloader.kt
package at.shockbytes.dante.network

import at.shockbytes.dante.book.BookSuggestion
import io.reactivex.Observable

/**
 * Author:  Martin Macheiner
 * Date:    13.02.2017
 */
interface BookDownloader {

    fun downloadBook(isbn: String): Observable<BookSuggestion>

    companion object {

        const val MAX_FETCH_AMOUNT = 10
    }
}
